pkg/sypgp/sypgp.go

```diff
@@ -597,50 +584,109 @@ func SelectPrivKey(el openpgp.EntityList) (*openpgp.Entity, error) {
-	listLine := "%s\t%s\t%s\n"
+	numKeys := 0
+	longFmt := "%s\t%s\t%s\t%s\t%s\t%s\t"
+	shortFmt := "%s\t%s\t"
+	nameFmt := "%s\n"
 
 	retList := bytes.NewBuffer(nil)
 	tw := tabwriter.NewWriter(retList, 0, 0, 2, ' ', 0)
-	fmt.Fprintf(tw, listLine, "KEY ID", "BITS", "NAME/EMAIL")
+	if longOutput {
+		fmt.Fprintf(tw, longFmt, "FINGERPRINT", "ALGORITHM", "BITS", "CREATION DATE", "EXPIRATION DATE", "STATUS")
+	} else {
+		fmt.Fprintf(tw, shortFmt, "KEY ID", "BITS")
+	}
+	fmt.Fprintf(tw, nameFmt, "NAME/EMAIL")
 
-	key := strings.Split(mrString, "\n")
+	lines := strings.Split(mrString, "\n")
 
-	for _, k := range key {
-		nk := strings.Split(k, ":")
-		for _, n := range nk {
-			if n == "info" {
-				var err error
-				keyNum, err = strconv.Atoi(nk[2])
-				if err != nil {
-					return -1, nil, fmt.Errorf("unable to check key number")
+	first := true
+	gotName := false
+	for _, l := range lines {
+		fields := strings.Split(strings.TrimSpace(l), ":")
+		if fields[0] == "info" {
+			var err error
+			numKeys, err = strconv.Atoi(fields[2])
+			if err != nil {
+				return -1, nil, fmt.Errorf("unable to check number of keys")
+			}
+		} else if fields[0] == "pub" {
+			if !first {
+				if !gotName {
+					// there was a pub without uid; end line
+					fmt.Fprintf(tw, "\n")
 				}
+				// put a blank line between each key
+				fmt.Fprintf(tw, "\n")
 			}
-			if n == "pub" {
-				// The fingerprint is located at nk[1], and we only want the last 8 chars
-				fmt.Fprintf(tw, "%s\t", nk[1][len(nk[1])-8:])
-				// The key size (bits) is located at nk[3]
-				fmt.Fprintf(tw, "%s\t", nk[3])
-				count++
+			first = false
+			gotName = false
+			keyFingerprint := fields[1]
+			keyBits := fields[3]
+			if longOutput {
+				keyType, err := getEncryptionAlgorithmName(fields[2])
+				if err != nil {
+					return -1, nil, err
+				}
+				keyDateCreated := date(fields[4])
+				keyDateExpired := date(fields[5])
+
+				keyStatus := ""
+				if fields[6] == "r" {
+					keyStatus = "[revoked]"
+				} else if fields[6] == "d" {
+					keyStatus = "[disabled]"
+				} else if fields[6] == "e" {
+					keyStatus = "[expired]"
+				} else {
+					keyStatus = "[enabled]"
+				}
+
+				fmt.Fprintf(tw, longFmt, keyFingerprint, keyType, keyBits, keyDateCreated, keyDateExpired, keyStatus)
+
+			} else {
+				// Short output
+				// take only last 8 chars of fingerprint
+				fmt.Fprintf(tw, shortFmt, keyFingerprint[len(fields[1])-8:], keyBits)
 			}
-			if n == "uid" {
-				// And the key name/email is on nk[1]
-				fmt.Fprintf(tw, "%s\t\n\n", nk[1])
+			count++
+		} else if fields[0] == "uid" {
+			// And the key name/email is on fields[1]
+			// There may be more than one of these for each pub
+			if !gotName {
+				gotName = true
+			} else {
+				// indent as far as the pub line
+				if longOutput {
+					fmt.Fprintf(tw, longFmt, "", "", "", "", "", "")
+				} else {
+					fmt.Fprintf(tw, shortFmt, "", "")
+				}
 			}
+			name := fields[1]
+			name, _ = url.QueryUnescape(name)
```

Comment:
Should we catch this error

Reply:
I thought it would be better to not catch that error.  Instead I just let it default to be the undecoded value if decode didn't work.  The Sylabs key server doesn't do the encoding, and if someone put a `%` into their key's uid for some reason it could trip up a decode.

Reply:
However, I hadn't actually tested that, now I have, and found this method didn't work.  I assumed that name would be left untouch if there was an error, but it gets turned into a blank string.  I put it in new code to make it work.  Test with
```
apptainer key search -u https://keys.sylabs.io Dykstra
```